### PR TITLE
Support of GPUs

### PR DESCRIPTION
**Pull request by Smartappli**

### CODE REVIEW

File: docker/Dockerfile.dev [hunk 1]
```diff
@@ -17,6 +17,7 @@ ENV NODE_ENV='development'
 
 # Install dependencies
 RUN apt-get update \
+    && apt-get upgrade -y \
```
Comment (gaby) [inline docker/Dockerfile.dev:20]: Remove this

File: docker/Dockerfile [hunk 2]
```diff
@@ -34,6 +34,7 @@ COPY vendor/requirements.txt /usr/src/app/requirements.txt
 
 # Install api dependencies
 RUN apt-get update \
+    && apt-get upgrade -y \
```
Comment (gaby) [inline docker/Dockerfile:37]: Remove this

File: docker-compose.gpu.yml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+services:
```
Comment (gaby) [inline docker-compose.gpu.yml:1]: This compose is not using gpus

File: docker/Dockerfile.gpu [hunk 4]
```diff
@@ -34,7 +38,9 @@ COPY vendor/requirements.txt /usr/src/app/requirements.txt
 
 # Install api dependencies
 RUN apt-get update \
-    && apt-get install -y --no-install-recommends dumb-init \
+    && apt-get upgrade -y \
```
Comment (gaby) [inline docker/Dockerfile.gpu:41]: Remove

**Review by gaby — CHANGES_REQUESTED** [review]

File: docker-compose.gpu.yml (hunk 3 above)
Comment (Smartappli) [reply docker-compose.gpu.yml:1]: What exactly do you want me to change?

File: docker/Dockerfile (hunk 2 above)
Comment (Smartappli) [reply docker/Dockerfile]: done

File: docker/Dockerfile.gpu (hunk 4 above)
Comment (Smartappli) [reply docker/Dockerfile.gpu]: done

File: docker/Dockerfile.dev (hunk 1 above)
Comment (Smartappli) [reply docker/Dockerfile.dev]: done

**Comment by Smartappli** [comment]
Can you help me with the docker compose gpu I don't know what to do.

**Comment by gaby** [comment]
@Smartappli Can you fix the merge conflict, will merge after that. Thanks! 💪

**Comment by Smartappli** [comment]
How do I resolve the conflict?

**Comment by Smartappli** [comment]
> @Smartappli Can you fix the merge conflict, will merge after that. Thanks! s💪

@gaby done

**Comment by gaby** [comment]
@Smartappli CI is failing

**Comment by kraoc** [comment]
I tried and made some changes:

**Dockerfile.gpu**
Before ```ENV NODE_ENV='production'``` need to add ```FROM debian:bookworm```

Also need to update all **pip install** to **pip install --break-system-packages** everywhere.

Image build is now ok but Serge still not seem to use gpu.

**Comment by gaby** [comment]
@kraoc llama.cpp has to be told to use gpu, everything is setup not to. This is just prep-work for the CI/project structure before adding gpu.

Not everyone has a gpu so Serge has to support all 3 cases: cpu, gpu, cpu/gpu.

File: docker/Dockerfile.gpu [hunk 5]
```diff
@@ -34,9 +40,11 @@ COPY vendor/requirements.txt /usr/src/app/requirements.txt
 
 # Install api dependencies
 RUN apt-get update \
-    && apt-get install -y --no-install-recommends dumb-init \
+    && apt-get install -y --no-install-recommends dumb-init python${PYTHON_VER} python${PYTHON_VER}-dev python3-pip python-is-python3 nvidia-modprobe \
+    && pip install --upgrade --no-cache-dir pip setuptools wheel \
     && pip install --no-cache-dir ./api \
     && pip install -r /usr/src/app/requirements.txt \
+    && pip install --break-system-packages everywhere \
```
Comment (gaby) [inline docker/Dockerfile.gpu:47]: Never seen this before

File: docker/Dockerfile.gpu [hunk 6]
```diff
@@ -14,9 +22,7 @@ COPY ./web /usr/src/app/web/
 WORKDIR /usr/src/app/web/
 RUN npm run build
 
-# ---------------------------------------
-# Runtime environment
-FROM python:3.11-slim-bookworm as release
+FROM debian:bookworm
```
Comment (gaby) [inline docker/Dockerfile.gpu:25]: This will remove all the previous stuff, including CUDA

**Comment by gaby** [comment]
@Smartappli Any updates on this?

**Pull request closed without merging**